File: 구현/1213.cpp
#include <bits/stdc++.h>
using namespace std;
int cnt[26], flag;
string s, ret;
char mid;
int main() {
    cin >> s;
    for (char c : s) cnt[(int)c]++;
    for (int i = 'Z'; i >= 'A'; i--) {
        if(cnt[i] & 1) {
            mid = char(i);
            flag++;
            cnt[i]--;
        } 
        if(flag == 2) {
            cout << "I'm Sorry Hansoo" << "\n";
            exit(0);
        }
        for (int j = 0; j < cnt[i]; j += 2) {
            ret = char(i) + ret;
            ret += char(i);
        }       
    }

    if(mid) ret.insert(ret.begin() + ret.size() / 2, mid);
    cout << ret << "\n";
    return 0;
}

File: 구현/1213.js
const [input] = ['ABCD'];

const alpha = Array(26).fill(0);
const asciiOfA = 'A'.charCodeAt();
for (let c of input) {
  alpha[c.charCodeAt(0) - asciiOfA]++;
}

let flag = 0;     
let left = '';
let mid = '';

for (let i = 0; i <= 25; i++) {
  if (alpha[i] & 1) {
    mid = String.fromCharCode(i + asciiOfA);
    flag++;
    alpha[i]--;
  }
  
  if (flag >= 2) {
    break;
  }

  for (let j = 0; j < Math.round(alpha[i] / 2); j++) {
    left += String.fromCharCode(i + asciiOfA);
  }
}

if (flag >= 2) {
  console.log("I'm Sorry Hansoo");
} else {
  console.log(left + mid + left.split('').reverse().join(''));  
}
